### PR TITLE
LDAP security realm - iterability

### PR DESCRIPTION
**Pull request by honza889**
* implemented Realm Identity Iterator
* organized private methods:
 * createSearchControls() moved outside of the identity (to be available for iterator)
 * principalName param of getIdentity() removed - redundant

### CODE REVIEW

**Comment by honza889** [comment]
Hmm... one test added and build on windows broken... looks like teamcity problem..

File: src/main/java/org/wildfly/security/auth/provider/ldap/LdapSecurityRealm.java [hunk 1]
```diff
@@ -115,7 +115,46 @@ public ModifiableRealmIdentity getRealmIdentityForUpdate(String name, final Prin
...
+        try {
+            context = dirContextFactory.obtainDirContext(null);
+
+            result = context.search(identityMapping.searchDn, identityMapping.iteratorFilter,
+                    identityMapping.iteratorFilterArgs, createSearchControls(identityMapping.rdnIdentifier));
+
+        } catch (NamingException e) {
+            throw log.ldapRealmIdentitySearchFailed(e);
+        } finally {
+            dirContextFactory.returnContext(context);
```
Comment (darranl) [inline src/main/java/org/wildfly/security/auth/provider/ldap/LdapSecurityRealm.java]: Is this definitely Ok that the context is returned but we are still working with the results?  Are the results eagerly loaded?

Comment (honza889) [inline src/main/java/org/wildfly/security/auth/provider/ldap/LdapSecurityRealm.java]: hmm... I have test it and it looks it is lazy. I will rewrite it.

**Comment by honza889** [comment]
I forgot mention, this PR is prepared for review now.

File: src/main/java/org/wildfly/security/auth/provider/ldap/LdapSecurityRealm.java [hunk 2]
```diff
@@ -115,7 +115,35 @@ public ModifiableRealmIdentity getRealmIdentityForUpdate(String name, final Prin
...
+            NamingEnumeration<SearchResult> result = context.search(identityMapping.searchDn, identityMapping.iteratorFilter,
+                    identityMapping.iteratorFilterArgs, createSearchControls(identityMapping.rdnIdentifier));
+
+            while(result.hasMore()) {
+                SearchResult entry = result.next();
+                String name = (String) entry.getAttributes().get(identityMapping.rdnIdentifier).get();
+                identifiers.add(getRealmIdentityForUpdate(name, null, null));
+            }
+
+            result.close();
```
Comment (pedroigor) [inline src/main/java/org/wildfly/security/auth/provider/ldap/LdapSecurityRealm.java]: I think you should close the result in the finally block as well.

**Comment by honza889** [comment]
* Added finally block for SearchResult closing
* After discussion with Darran revised LDAP DirContext returning - should be returned only if was obtained successfuly (should not be returned if exception occure in obtaining)

**Pull request closed without merging**